apps/backoffice-v2/src/domains/alerts/hooks/mutations/useAssignAlertsMutation/useAssignAlertsMutation.tsx

```diff
@@ -0,0 +1,80 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useUsersQuery } from '@/domains/users/hooks/queries/useUsersQuery/useUsersQuery';
+import toast from 'react-hot-toast';
+import { t } from 'i18next';
+import { assignAlertsByIds } from '@/domains/alerts/fetchers';
+
+const getToastActionAndContext = ({
+  assigneeId,
+  assigneeName,
+  isAssignedToMe,
+}: {
+  assigneeId: string | null;
+  assigneeName: string | null;
+  isAssignedToMe: boolean;
+}) => {
+  const action = assigneeId ? 'assign_alerts' : 'unassign_alerts';
+  const context = assigneeId
+    ? {
+        assignee: isAssignedToMe ? 'me' : assigneeName,
+      }
+    : {};
+
+  return { action, context } as const;
+};
+
+export const useAssignAlertsByIdsMutation = ({
+  onSuccess,
+}: {
+  onSuccess?: <TData>(
+    data: TData,
+    { assigneeId, isAssignedToMe }: { assigneeId: string | null; isAssignedToMe: boolean },
+    { assigneeName }: { assigneeName: string | null },
+  ) => void;
+} = {}) => {
+  const queryClient = useQueryClient();
+  const { data: users } = useUsersQuery();
+
+  return useMutation({
+    mutationFn: ({
+      assigneeId,
+      alertIds,
+    }: {
+      assigneeId: string | null;
+      alertIds: string[];
+      isAssignedToMe: boolean;
```

Comment:
why does it matter?
in the component which is assign to the same assigneeId you just set another label ... 
i think its redundant to keep such a property

Reply:
Lets talk about it in tech debt please add a task and specify this comment link in the description